pydrive2/fs.py

```diff
@@ -0,0 +1,476 @@
+import errno
+import io
+import logging
+import os
+import posixpath
+import threading
+from collections import defaultdict
+
+from fsspec.spec import AbstractFileSystem
+from funcy import cached_property, retry, wrap_prop, wrap_with
+from funcy.py3 import cat
+from tqdm.utils import CallbackIOWrapper
+
+from pydrive2.drive import GoogleDrive
+from pydrive2.fs_utils import IterStream
+
+logger = logging.getLogger(__name__)
+
+FOLDER_MIME_TYPE = "application/vnd.google-apps.folder"
+
+
+def _gdrive_retry(func):
```

Comment:
I think there is a very similar implementation in tests for this. It's not a big deal, but would be nice to consolidate them (move into some utils?)

Reply:
Seems like the behavior is different, even though they can be unified I think it's better to just keep this as is for preventing any kind of behavioral change in either the fs or the tests. (https://github.com/iterative/PyDrive2/pull/119/files#r682487839)
```py
# 15 tries, start at 0.5s, multiply by golden ratio, cap at 20s
@retry(15, PyDriveRetriableError, timeout=lambda a: min(0.5 * 1.618 ** a, 20))
def pydrive_retry(call, *args, **kwargs):
    try:
        result = call(*args, **kwargs)
    except ApiRequestError as exception:
        if exception.error["code"] in [403, 500, 502, 503, 504]:
            raise PyDriveRetriableError("Google API request failed")
        raise
    return result
```

Reply:
👍